homeassistant/components/airly/air_quality.py

```diff
@@ -36,88 +44,75 @@
 PARALLEL_UPDATES = 1
 
 
-async def async_setup_entry(hass, config_entry, async_add_entities):
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
     """Set up Airly air_quality entity based on a config entry."""
-    name = config_entry.data[CONF_NAME]
+    name = entry.data[CONF_NAME]
 
-    coordinator = hass.data[DOMAIN][config_entry.entry_id]
+    coordinator = hass.data[DOMAIN][entry.entry_id]
 
     async_add_entities([AirlyAirQuality(coordinator, name)], False)
 
 
-def round_state(func):
-    """Round state."""
-
-    def _decorator(self):
-        res = func(self)
-        if isinstance(res, float):
-            return round(res)
-        return res
-
-    return _decorator
-
-
 class AirlyAirQuality(CoordinatorEntity, AirQualityEntity):
     """Define an Airly air quality."""
 
-    def __init__(self, coordinator, name):
+    def __init__(self, coordinator: AirlyDataUpdateCoordinator, name: str) -> None:
         """Initialize."""
         super().__init__(coordinator)
         self._name = name
         self._icon = "mdi:blur"
 
     @property
-    def name(self):
+    def name(self) -> str:
         """Return the name."""
         return self._name
 
     @property
-    def icon(self):
+    def icon(self) -> str:
         """Return the icon."""
         return self._icon
 
     @property
-    @round_state
-    def air_quality_index(self):
+    def air_quality_index(self) -> float | None:
         """Return the air quality index."""
-        return self.coordinator.data[ATTR_API_CAQI]
+        return round_state(self.coordinator.data[ATTR_API_CAQI])
 
     @property
-    @round_state
-    def particulate_matter_2_5(self):
+    def particulate_matter_2_5(self) -> float | None:
         """Return the particulate matter 2.5 level."""
-        return self.coordinator.data.get(ATTR_API_PM25)
+        return round_state(self.coordinator.data.get(ATTR_API_PM25))
 
     @property
-    @round_state
-    def particulate_matter_10(self):
+    def particulate_matter_10(self) -> float | None:
         """Return the particulate matter 10 level."""
-        return self.coordinator.data.get(ATTR_API_PM10)
+        return round_state(self.coordinator.data.get(ATTR_API_PM10))
 
     @property
-    def attribution(self):
+    def attribution(self) -> str:
         """Return the attribution."""
         return ATTRIBUTION
 
     @property
-    def unique_id(self):
+    def unique_id(self) -> str:
         """Return a unique_id for this entity."""
-        return f"{self.coordinator.latitude}-{self.coordinator.longitude}"
+        return f"{self.coordinator.latitude}-{self.coordinator.longitude}"  # type: ignore[attr-defined]
 
     @property
-    def device_info(self):
+    def device_info(self) -> dict[str, Any]:
```

Comment:
This changed recently.
```suggestion
    def device_info(self) -> DeviceInfo:
```